### PR TITLE
fix(depends): Avoid digits in archive file extension

### PR DESCRIPTION
**Pull request by issaclin32**
* This avoids extraction error for 7-zip split archives like `xxx.7z.001`, `xxx.7z.002`, ....

for example:

```
"url": [
    "https://someapp.com/files/some-app.7z.001",
    "https://someapp.com/files/some-app.7z.002"
]
```

In this case. `some-app.7z.001` extracts properly but `some-app.7z.002` is treated as another independent archive, resulting in extraction error.

* This fix would be useful when we need to split files in ScoopInstaller/Binary (into multiple 7z archives) to avoid using Git LFS.

### CODE REVIEW

**Comment by issaclin32** [comment]
By the way, this fix would be useful when we need to split files in `ScoopInstaller/Binary` (into multiple 7z archives) to avoid using Git LFS.

**Comment by niheaven** [comment]
After this is applied, are `.00x` removed after 7z extracted the files?

**Comment by issaclin32** [comment]
> After this is applied, are `.00x` removed after 7z extracted the files?

Missed that. Thanks for reminding me.

File: lib/decompress.ps1 [hunk 1]
```diff
@@ -65,6 +65,11 @@ function Expand-7zipArchive {
     if ($Removal) {
         # Remove original archive file
         Remove-Item $Path -Force
+        if ([System.IO.Path]::GetExtension($Path) -eq '.001') {
+            # Remove remaining 7-zip archive parts
+            $FileNameWithoutExt = [System.IO.Path]::GetFileNameWithoutExtension($Path)
+            Get-ChildItem $([System.IO.Path]::GetDirectoryName($Path)) | Where {$_ -match ([Regex]::Escape($FileNameWithoutExt)+'\.\d{3}$')} | Remove-Item
+        }
```
Comment (niheaven) [inline lib/decompress.ps1:72]:
```suggestion
        if (($Path -replace '.*\.([^\.]*)$', '$1') -eq '001') {
            # Remove remaining 7-zip archive parts
            Get-ChildItem "$($Path -replace '\.[^\.]*$', '').???" | Remove-Item -Force
        } else {
            Remove-Item $Path -Force
        }
```

**Review by niheaven — COMMENTED** [review]
1. Using `-replace` may be more readable
2. I've updated the test scripts and test cases, please review them

**Comment by issaclin32** [comment]
> 2. I've updated the test scripts and test cases, please review them

The test scripts & cases works fine for me. Thank you.

**Pull request merged**